app/models/openscap_result.rb

```diff
@@ -0,0 +1,77 @@
+class OpenscapResult < ApplicationRecord
+  belongs_to :container_image
+  has_one    :binary_blob,           :dependent => :destroy, :autosave => true, :as => :resource
+  has_many   :openscap_rule_results, :dependent => :destroy, :autosave => true
+
+  def self.openscap_available?
+    # needed only for travis
+    require 'openscap'
+    require 'openscap/ds/arf'
+    require 'openscap/xccdf/benchmark'
+    true
+  rescue LoadError
+    false
+  end
+
+  def html
+    raw = binary_blob.try(:binary)
+    return unless raw
+    with_openscap_arf(raw) do |arf|
+      ascii8bit_to_utf8(arf.html)
+    end
+  end
+
+  def raw=(value)
+    self.binary_blob = BinaryBlob.new(:name => 'openscap_compliance_arf', :data_type => 'XML')
+    binary_blob.binary = value
+    with_openscap_objects(binary_blob.binary) do |rule_results, benchmark_items|
+      create_results(rule_results, benchmark_items)
```

Comment:
Hrm, should this actually create a whole new set of results each time `raw=` is called?

For instance, if I have the following calls:

```ruby

result = OpenscapResult.first
result.raw = a_binary_blob       # two rule results in this blob
result.raw = another_binary_blob # three rule results in this blob
```

It seems like I'm gonna have the following:

```ruby
result.raw == another_binary_blob

# two from the first assignment and three from the second
result.openscap_rule_results.size == 5 
```

Maybe I read something wrong though?